embedded/src/main/java/org/wildfly/core/embedded/EmbeddedStandAloneServerFactory.java

```diff
@@ -0,0 +1,417 @@
+
+        try {
+            File configDir = new File(tempRoot, "config");
+            Files.createDirectory(configDir.toPath());
+            File dataDir = new File(tempRoot, "data");
+            Files.createDirectory(dataDir.toPath());
+            // For jboss.server.deployment.scanner.default
+            File deploymentsDir = new File(tempRoot, "deployments");
+            Files.createDirectory(deploymentsDir.toPath());
+
+            copyDirectory(originalConfigDir, configDir);
+            if (originalDataDir.exists()) {
+                copyDirectory(originalDataDir, dataDir);
+            }
+
+            props.put(ServerEnvironment.SERVER_BASE_DIR, tempRoot.getAbsolutePath());
+            props.put(ServerEnvironment.SERVER_CONFIG_DIR, configDir.getAbsolutePath());
+            props.put(ServerEnvironment.SERVER_DATA_DIR, dataDir.getAbsolutePath());
+        }  catch (IOException e) {
+            throw EmbeddedLogger.ROOT_LOGGER.cannotSetupEmbeddedServer(e);
+        }
+
+    }
+
+    private static File getFileUnderAsRoot(File jbossHomeDir, Properties props, String propName, String relativeLocation, boolean mustExist) {
+        String prop = props.getProperty(propName, null);
+        if (prop == null) {
+            prop = props.getProperty(ServerEnvironment.SERVER_BASE_DIR, null);
+            if (prop == null) {
+                File dir = new File(jbossHomeDir, "standalone" + File.separator + relativeLocation);
+                if (mustExist && (!dir.exists() || !dir.isDirectory())) {
+                    throw ServerLogger.ROOT_LOGGER.embeddedServerDirectoryNotFound("standalone" + File.separator + relativeLocation, jbossHomeDir.getAbsolutePath());
+                }
+                return dir;
+            } else {
+                File server = new File(prop);
+                validateDirectory(ServerEnvironment.SERVER_BASE_DIR, server);
+                return new File(server, relativeLocation);
+            }
+        } else {
+            File dir = new File(prop);
+            validateDirectory(ServerEnvironment.SERVER_CONFIG_DIR, dir);
+            return dir;
+        }
+
+    }
+
+    private static File getTempRoot(Properties props) {
+        String tempRoot = props.getProperty(JBOSS_EMBEDDED_ROOT, null);
+        if (tempRoot == null) {
+            return null;
+        }
+
+        try {
+            File root = new File(tempRoot);
+            if (!root.exists()) {
+                //Attempt to try to create the directory, in case something like target/embedded was specified
+                Files.createDirectories(root.toPath());
+            }
+            validateDirectory("jboss.test.clean.root", root);
+            root = new File(root, "configs");
+            Files.createDirectories(root.toPath());
+            SimpleDateFormat format = new SimpleDateFormat("yyyyMMddHHmmssSSS");
+            root = new File(root, format.format(new Date()));
+            Files.createDirectory(root.toPath());
```

Comment:
Yeah, doesn't really matter all that much. I just cringe a little when I see `SimpleDateFormat`, but really it's not like it will be used a lot here so it's fine :)